core/modules/replica/ReplicationRequest.h

```diff
@@ -0,0 +1,201 @@
+     *                          at a destination of the chunk
+     * @param sourceWorker    - the identifier of a worker node at a source of the chunk
+     * @param database        - the name of a database
+     * @param chunk           - the number of a chunk to replicate (implies all relevant tables)
+     * @param onFinish        - an optional callback function to be called upon a completion of the request.
+     * @param priority        - a priority level of the request
+     * @param keepTracking    - keep tracking the request before it finishes or fails
+     * @param allowDuplicate  - follow a previously made request if the current one duplicates it
+     * @param messenger       - worker messenging service
+     */
+    static pointer create(ServiceProvider::pointer const& serviceProvider,
+                          boost::asio::io_service& io_service,
+                          std::string const& worker,
+                          std::string const& sourceWorker,
+                          std::string const& database,
+                          unsigned int  chunk,
+                          callback_type onFinish,
+                          int  priority,
+                          bool keepTracking,
+                          bool allowDuplicate,
+                          std::shared_ptr<Messenger> const& messenger);
+
+private:
+
+    /**
+     * Construct the request with the pointer to the services provider.
+     */
+    ReplicationRequestM(ServiceProvider::pointer const& serviceProvider,
+                        boost::asio::io_service& io_service,
+                        std::string const& worker,
+                        std::string const& sourceWorker,
+                        std::string const& database,
+                        unsigned int  chunk,
+                        callback_type onFinish,
+                        int  priority,
+                        bool keepTracking,
+                        bool  allowDuplicate,
+                        std::shared_ptr<Messenger> const& messenger);
+
+    /**
+      * Implement the method declared in the base class
+      *
+      * @see Request::startImpl()
+      */
+    void startImpl() final;
+
+    /// Start the timer before attempting the previously failed
+    /// or successfull (if a status check is needed) step.
+    void wait();
+
+    /// Callback handler for the asynchronious operation
+    void awaken(boost::system::error_code const& ec);
+
+    /// Send the serialized content of the buffer to a worker
+    void send();
+
+    /// Process the completion of the requested operation
+    void analyze(bool success,
+                 proto::ReplicationResponseReplicate const& message);
+
+    /**
+     * Notifying a party which initiated the request.
+     *
+     * This method implements the corresponing virtual method defined
+     * bu the base class.
+     */
+    void notify() final;
+
+private:
+
+    // Parameters of the object

```

Comment:
#deleteme

Reply:
Here is an illustration why I keep putting this line class definitions:
`
private:

    // Parameters of the object

    ServiceProvider::Ptr _serviceProvider;
    std::string _workerName;

    // Cached parameters of the worker

    WorkerProcessor _processor;

    WorkerInfo const& _workerInfo;

    // The mutable state of the object

    boost::asio::io_service        _io_service;
    boost::asio::ip::tcp::acceptor _acceptor;
};
`

Reply:
The idea here is to group parameters according to their origin (constructor: "parameters of the object"), post initialization ("Cached parameters...") and the mutable state. I'm not sure of this approach makes sense to everyone (hence we have this code review). However, I'm happy to drop this comment where this is the only comment in this context.